std/str.zig

```diff
@@ -10,6 +11,15 @@ pub fn slice_eql(T: type)(a: []const T, b: []const T) -> bool {
     return true;
 }
 
+pub fn join(T: type)(a: []T, b: []T) -> []T {
+    var result : [a.len + b.len]T = undefined;
```

Comment:
a.len and b.len are runtime values. this will cause result to be allocated on the stack.